sys/shell/shell.c

```diff
@@ -227,14 +235,36 @@ static void handle_input_line(const shell_command_t *command_list, char *line)
     }
 }
 
+/**
+ * Read a single line from standard input into a buffer.
+ *
+ * In addition to copying characters, this routine echoes the line back to
+ * stdout and also supports primitive line editing.
+ *
+ * If the input line is too long, the input will still be consumed until the end
+ * to prevent the next line from containing garbage.
+ *
+ * @param   buf     Buffer where the input will be placed.
+ * @param   size    Size of the buffer. The maximum line length will be one less
+ *                  than size, to accommodate for the null terminator.
+ *                  The minimum buffer size is 1.
+ *
+ * @return  length of the read line, excluding the terminator, if reading was
+ *          successful.
+ * @return  EOF, if the if the end of the input stream is reached.
+ * @return  -READLINE_TOOLONG if the buffer size was exceeded
+ */
 static int readline(char *buf, size_t size)
 {
-    char *line_buf_ptr = buf;
+    int curr_pos = 0;
```

Comment:
maybe use `unsigned` here? 

Also I don't like this half-abbreviated variable name, if you want it short why not simply use `pos` or `bufpos` to indicate the what and where.

Reply:
The current position is used as return value, so it I use an unsigned int there I need to do an explicit cast in line 290.

Reply:
The file is already using abbreviated names `buf`, `line_buf_ptr`. This does not change the style.

Reply:
fine, leave as is